operator/helm/helm.go

```diff
@@ -0,0 +1,324 @@
+	// TODO: implement.
+	return nil
+}
+
+// Get satisfies the Get interface. It gets a Helm installation using its name.
+func (c *Client) Get(name string, opts GetOpts) (*Release, error) {
+	namespace := opts.Namespace.ValueOrDefault()
+	cfg, err := c.config(namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Helm release %q: %w", name, err)
+	}
+	client := action.NewGet(cfg)
+	res, err := client.Run(name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Helm release %q: %w", name, err)
+	}
+	return &Release{res}, nil
+}
+
+func (c *Client) config(namespace string) (*action.Configuration, error) {
+	restGetter := kube.GetConfig(defaultKubeConfig, defaultContext, namespace)
+	debug := func(format string, v ...interface{}) {
+		// TODO(f0rmiga): provide a logic for Helm debugging logs.
+	}
+	cfg := &action.Configuration{}
+	if err := cfg.Init(restGetter, namespace, driver, debug); err != nil {
+		return nil, fmt.Errorf("failed to provide action configuration: %w", err)
+	}
+	return cfg, nil
+}
+
+// ChartCache manages the local chart cache.
+type ChartCache struct {
+	cachePath string
+
+	osStat     func(name string) (os.FileInfo, error)
+	osOpenFile func(name string, flag int, perm os.FileMode) (*os.File, error)
+	httpGetter HTTPGetter
+
+	mutex sync.Mutex
+}
+
+// NewChartCache constructs a new ChartCache.
+func NewChartCache(cachePath string) *ChartCache {
+	return &ChartCache{
+		cachePath:  cachePath,
+		osStat:     os.Stat,
+		osOpenFile: os.OpenFile,
+		httpGetter: http.DefaultClient,
+	}
+}
+
+// Fetch downloads a chart tarball if needed, adding it to the cache. The
+// returned tarball is always a handle to the cached file.
+func (cc *ChartCache) Fetch(chartInfo ChartInfo) (io.ReadCloser, error) {
+	fileName := fmt.Sprintf("%s.tgz", chartInfo.SHA256Sum)
+	filePath := path.Join(cc.cachePath, fileName)
+	if _, err := cc.osStat(filePath); err == nil {
+		r, err := cc.osOpenFile(filePath, os.O_RDONLY, 0)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch chart from %q: %w", chartInfo.URL, err)
+		}
+		return r, nil
+	} else if !os.IsNotExist(err) {
+		// The error exists and it's not of the type non-existent file. I.e. an
+		// unexpected error.
+		return nil, fmt.Errorf("failed to fetch chart from %q: %w", chartInfo.URL, err)
+	}
+
+	res, err := cc.httpGetter.Get(chartInfo.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch chart from %q: %w", chartInfo.URL, err)
+	}
+	defer res.Body.Close()
+	if err := cc.cache(filePath, res.Body, chartInfo.SHA256Sum); err != nil {
+		return nil, fmt.Errorf("failed to fetch chart from %q: %w", chartInfo.URL, err)
+	}
+
+	r, err := cc.osOpenFile(filePath, os.O_RDONLY, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch chart from %q: %w", chartInfo.URL, err)
+	}
+	return r, nil
+}
+
+func (cc *ChartCache) cache(filePath string, data io.Reader, sha256sum string) error {
+	cc.mutex.Lock()
```

Comment:
I guess this mutex is to prevent two instances from writing the same cache file? But we're not using the mutex in `Fetch`, so we could (assuming a currently-running bad `cache` call) end up erroring out because the `os.Stat` passed, then the (bad) file got deleted, and then `osOpenFile` errors.

It might be better to do the write-out-temp-file-then-rename thing here instead.  Probably still needs a mutex around (if the temp file name is deterministic) though.

Reply:
Fixed with f339203.